File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

// FIXME: This example is incomplete

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your RGB LED Button Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var rlb = new Tinkerforge.BrickletRGBLEDButton(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

// Register button state changed callback
rlb.on(Tinkerforge.BrickletRGBLEDButton.CALLBACK_BUTTON_STATE_CHANGED,
    // Callback function for button state changed callback
    function (state) {
        console.log('State: ' + state);
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

# FIXME: This example is incomplete

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletRGBLEDButton;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your RGB LED Button Bricklet

# Callback subroutine for button state changed callback
sub cb_button_state_changed
{
    my ($state) = @_;

    print "State: $state\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $rlb = Tinkerforge::BrickletRGBLEDButton->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register button state changed callback to subroutine cb_button_state_changed
$rlb->register_callback($rlb->CALLBACK_BUTTON_STATE_CHANGED, 'cb_button_state_changed');

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
